src/Promitor.Scraper.Tests.Unit/Serialization/v1/Core/AzureMetricConfigurationDeserializerTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.ComponentModel;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Promitor.Core.Scraping.Configuration.Serialization;
+using Promitor.Core.Scraping.Configuration.Serialization.v1.Core;
+using Promitor.Core.Scraping.Configuration.Serialization.v1.Model;
+using Xunit;
+using YamlDotNet.RepresentationModel;
+
+namespace Promitor.Scraper.Tests.Unit.Serialization.v1.Core
+{
+    [Category("Unit")]
+    public class AzureMetricConfigurationDeserializerTests
+    {
+        private readonly AzureMetricConfigurationDeserializer _deserializer;
+        private readonly Mock<IDeserializer<MetricAggregationV1>> _aggregationDeserializer;
+
+        public AzureMetricConfigurationDeserializerTests()
+        {
+            _aggregationDeserializer = new Mock<IDeserializer<MetricAggregationV1>>();
```

Comment:
Is there a reason why we mock it and not use the real one instead?

Reply:
So that we're testing a single deserializer at once. If we don't mock it, we're going into the realms of an integration test because we'll be testing multiple deserializers at once, making these tests more brittle.

Reply:
I was more concerned for all deserializers to work together and no longer work but we have the bigger full test so that's covered already.

Resolving!